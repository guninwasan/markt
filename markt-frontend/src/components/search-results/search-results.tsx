import { useSearchParams } from "react-router-dom";
import { ListingContainer } from "../listing-container";
import { useEffect, useState } from "react";
import { dummyData1, dummyData2 } from "../dummy-listing-data";
import {
  NoResultsContainer,
  NoResultsHeading,
  NoResultsText,
  SearchResultsContainer,
  ResultsContainer,
} from "./search-results.styles";

const SearchResults = () => {
  const [searchParams] = useSearchParams();
  const keywords = searchParams.get("keywords");

  const [searchResults, setSearchResults] = useState<any>([]);

  useEffect(() => {
    console.log("searching for: ", keywords);
    // we need to add an API call here to fetch search results
    setSearchResults([...dummyData1, ...dummyData2]);
  }, [keywords]);

  return (
    <>
      {searchResults.length === 0 ? (
        <NoResultsContainer>
          <NoResultsHeading>No Results Found</NoResultsHeading>
          <NoResultsText>
            There were no results found for your search. Try searching for
            something else or check back later.
            <br />
            <br />
            Meanwhile, feel free to browse through our featured listings below.
          </NoResultsText>
        </NoResultsContainer>
      ) : (
        <ResultsContainer>Your search results for: {keywords}</ResultsContainer>
      )}
      <SearchResultsContainer>
        {searchResults.map((result: any) => (
          <ListingContainer
            key={result.id}
            id={result.id}
            title={result.title}
            price={result.price}
            image={result.image}
            condition="new"
            location="location"
          />
        ))}
      </SearchResultsContainer>
    </>
  );
};

export { SearchResults };
