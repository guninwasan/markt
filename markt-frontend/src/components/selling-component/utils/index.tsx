export * from "./upload-image";
export * from "./valid-form-details";
