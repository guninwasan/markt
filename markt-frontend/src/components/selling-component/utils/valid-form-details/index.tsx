export * from "./valid-form-details";
