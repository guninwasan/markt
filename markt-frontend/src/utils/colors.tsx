const colors = {
  primary: "#b8d3e0",
  textBlack: "#333",
  black: "#333",
  lightGrey: "#d9d9d9",
  white: "#f9f9f9",
};

export { colors };
