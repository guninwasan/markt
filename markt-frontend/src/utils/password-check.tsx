const passwordCheck = (password: string) => {
  const isAtLeast8Chars = password.length >= 8;
  const hasLowerCase = /[a-z]/.test(password);
  const hasUpperCase = /[A-Z]/.test(password);
  const hasNumber = /\d/.test(password);
  const hasSpecialChar = /[@$!%*?&#-]/.test(password);

  return (
    isAtLeast8Chars &&
    hasLowerCase &&
    hasUpperCase &&
    hasNumber &&
    hasSpecialChar
  );
};

export { passwordCheck };
