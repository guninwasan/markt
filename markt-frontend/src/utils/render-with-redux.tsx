import { render } from "@testing-library/react";
import { BrowserRouter, MemoryRouter } from "react-router-dom";
import { Provider } from "react-redux";
import { store } from "../redux";

export const renderWithRedux = (component: React.ReactNode) => {
  return render(
    <Provider store={store}>
      <MemoryRouter>{component}</MemoryRouter>
    </Provider>
  );
};
