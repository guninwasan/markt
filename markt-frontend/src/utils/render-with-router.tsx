import { render } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";

export const renderWithRouter = (component: React.ReactNode) => {
  return render(<BrowserRouter>{component}</BrowserRouter>);
};
