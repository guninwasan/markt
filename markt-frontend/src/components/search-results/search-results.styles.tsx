import styled from "styled-components";
import { colors } from "../../utils";

const SearchResultsContainer = styled.div`
  display: flex;
  flex-wrap: wrap;
  gap: 2rem;
  margin: 2rem;
`;

const NoResultsContainer = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  width: 100%;
  text-align: center;
  flex-direction: column;
`;

const NoResultsHeading = styled.h1`
  font-size: 2rem;
  color: ${colors.textBlack};
`;

const NoResultsText = styled.p`
  font-size: 1rem;
  color: ${colors.textBlack};
`;

const ResultsContainer = styled.div`
  display: flex;
  flex-wrap: wrap;
  gap: 2rem;
  margin: 2rem;
  color: ${colors.textBlack};
`;

export {
  SearchResultsContainer,
  NoResultsContainer,
  NoResultsHeading,
  NoResultsText,
  ResultsContainer,
};
