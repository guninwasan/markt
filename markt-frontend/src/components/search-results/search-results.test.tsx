import React from "react";
import { render, screen } from "@testing-library/react";
import { SearchResults } from "./search-results";
import { useSearchParams } from "react-router-dom";
import { renderWithRouter } from "../../utils/render-with-router";

jest.mock("react-router-dom", () => ({
  ...jest.requireActual("react-router-dom"),
  useSearchParams: jest.fn(),
}));

describe("SearchResults", () => {
  const mockUseSearchParams = useSearchParams as jest.Mock;

  beforeEach(() => {
    mockUseSearchParams.mockReturnValue([
      {
        get: jest.fn().mockReturnValue("test"),
      },
    ]);
  });

  it("should render search results", () => {
    renderWithRouter(<SearchResults />);
    expect(
      screen.getByText("Your search results for: test")
    ).toBeInTheDocument();
  });

  it("should match snapshot", () => {
    const { container } = renderWithRouter(<SearchResults />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
