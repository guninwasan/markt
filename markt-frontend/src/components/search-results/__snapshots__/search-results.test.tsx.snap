// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchResults should match snapshot 1`] = `
<div
  class="sc-cHqXqK dGpFxA"
>
  Your search results for: 
  test
</div>
`;
